hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -70,6 +70,10 @@
   private int type = 0;
   private Map<String, HoodieMetadataFileInfo> filesystemMetadata = null;
 
+  public HoodieMetadataPayload(GenericRecord record, Comparable<?> orderingVal) {
```

Comment:
would all existing payloads be forced to implement this new constructor?

Reply:
> would all existing payloads be forced to implement this new constructor?

Yes.